pandas/core/arrays/arrow/array.py

```diff
@@ -506,6 +506,19 @@ def __len__(self) -> int:
         """
         return len(self._data)
 
+    def __contains__(self, key) -> bool:
+        # https://github.com/pandas-dev/pandas/pull/51307#issuecomment-1426372604
+        if isna(key) and key is not self.dtype.na_value:
+            if self.dtype.kind == "f" and lib.is_float(key) and isna(key):
+                res = (pc.is_null(self._data, nan_is_null=True) & ~self.isna()).any()
+                return bool(res)
+
+            # e.g. date or timestamp types we do not allow None here to match pd.NA
```

Comment:
So to confirm, this makes `pd.NaT in array` False (I think that's reasonable)

Reply:
correct